packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -520,83 +519,96 @@ class HeapTreeViewState extends State<HeapTree>
-                  controller.groupByTreeTable.dataRoots.every((element) {
-                    element.collapseCascading();
-                    return true;
-                  });
-                  if (controller.instanceFieldsTreeTable != null) {
-                    // We're collapsing close the fields table.
-                    controller.selectedLeaf = null;
-                  }
-                  // All nodes collapsed - signal tree state changed.
-                  controller.treeChanged();
-                }
-              }
-            : null,
-        child: const Icon(
-          Icons.vertical_align_top,
-          size: defaultIconSize,
-        ),
-      ),
-      FixedHeightOutlinedButton(
-        buttonKey: expandAllButtonKey,
-        tooltip: 'Expand All',
-        onPressed: snapshotDisplay is MemoryHeapTable
-            ? () {
-                MemoryScreen.gaAction(key: expandAllButtonKey);
-                if (snapshotDisplay is MemoryHeapTable) {
-                  controller.groupByTreeTable.dataRoots.every((element) {
-                    element.expandCascading();
-                    return true;
-                  });
-                }
-                // All nodes expanded - signal tree state  changed.
-                controller.treeChanged();
-              }
-            : null,
-        child: const Icon(
-          Icons.vertical_align_bottom,
-          size: defaultIconSize,
+    return Row(
+      children: [
+        FixedHeightOutlinedButton(
+          buttonKey: snapshotButtonKey,
+          tooltip: 'Take a memory profile snapshot',
+          onPressed: _isSnapshotRunning ? null : _takeHeapSnapshot,
+          child: const MaterialIconLabel(
+            Icons.camera,
+            'Take Heap Snapshot',
+          ),
         ),
-      ),
-      const SizedBox(width: defaultSpacing),
-      Container(
-        width: 170,
-        child: SwitchListTile(
-          title: const Text('Treemap'),
-          value: controller.treeMapVisible.value,
-          onChanged: controller.snapshotByLibraryData != null
-              ? (value) {
-                  setState(() {
-                    controller.treeMapVisible.value = value;
-                  });
-                }
-              : null,
+        const SizedBox(width: defaultSpacing),
+        Row(
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            const Text('Treemap'),
+            // TODO(terry): Wrap in a ValueListenableBuilder to avoid calling setState.
+            Switch(
+              value: controller.treeMapVisible.value,
+              onChanged: controller.snapshotByLibraryData != null
+                  ? (value) {
+                      setState(() {
+                        controller.treeMapVisible.value = value;
+                      });
+                    }
+                  : null,
+            ),
+          ],
         ),
-      ),
-    ]);
+        const SizedBox(width: defaultSpacing),
```

Comment:
as a cleanup in a follow up PR, this can just be 
```
if (controller.treeMapVisible.value) ...[
  dropdown,
  expand all button,
  collapse all button,
],
```
and actually we'll want the entire snapshot controls row to be wrapped in a value listenable builder listening to the switch changes